Utility.cpp

```diff
@@ -113,3 +113,12 @@ void useAttackItem(Character* character, Item* item)
     }
 }
 
+void boostAndUpdateStats(std::unique_ptr<int>& initialStat, int& stat, float factor)
```

Comment:
This function is only used by `Character`. 
it doesn't need to exist outside of `Character.cpp`
Always try to limit the scope of functions and variables to the smallest scope possible.
Because it is confined to `Character.cpp`, it does not need a `factor` function argument.
Also, just pass `int&` instead of `std::unique_ptr<int>&` 
You're dereferencing 3 times.  
If you change the function argument, you'll only need to dereference once.
- [ ] revise